### ISSUE TITLE
How do I launch the app?  

### ISSUE THREAD

**Issue by roschler** [Hyperion-BT/PicoSwap#4]
I have VSCode installed with the Deno extension.  I did execute the "Deno: initialize" command from my workspace folder, after I cloned the Picoswap repository.  I also followed the Readme instructions all the way to the Deno caching instruction.  I then launch Deno and I do see it listening on port 9229 on my Linux box when I execute a "netstat -tulpn" command.  

However, when I try to access the app from Chrome using a URL like "http://localhost:9229/", I get a "Not found" error showing as the sole text in the result page, and no other information.  How do I start the app from VSCode?  

I found this launch configuration on the web (launch.json) for VSCode, and I tried putting that in my `.vscode` directory:
```json
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Deno",
            "type": "node",
            "request": "launch",
            "cwd": "${workspaceFolder}",
            "runtimeExecutable": "deno",
            "runtimeArgs": ["run", "--inspect-brk", "-A", "index.js"],
            "port": 9229
        }
    ]
}
```
But still no luck and once Deno is launched running that again results in an address in use error, so I don't think that launch.json is correct for Picoswap?  I looked through the Deno web site's `Getting Started` section but I did not see anything helpful.  How do I start this app?

**Comment by christianschmitz** [comment]
I don't how to do this from within VSCode. I just use the `http-server` command which you can install using `npm`.

```
sudo npm install -g http-server

http-server . --cors
```

The default port is 8080

Updating does require two actions though: save and refresh

As an alternative you can look into webpack-dev-server, which automatically refreshes when you save.

**Comment by roschler** [comment]
Thanks Christian.  I'll try that in the morning.  Can you tell me where/how Deno fits in during the runtime execution pipeline?  I believe that its purpose is to act as a real-time transpiler or interpreter for the custom script I see in the JS files? (e.g. - some of the exotic `import` statement notation I see in some files, etc.).  

What I really can't figure out is how it participates in the call chain starting from the HTTP server when that module executes a URL request against the code base.  I'm pretty sure that the `http-server` package doesn't know how to talk to Deno, right?  So how is Deno, which I see listening on port `9229` on my box, participating?

**Comment by christianschmitz** [comment]
Deno is only used for the IDE typechecking when importing modules directly from a https source.

If you create a dApp as a single page app using nodejs/react/webpack, then VSCode is smart enough to figure out the types out-of-the-box.

Picoswap is completely static, there is no build process (I might change it to use react/webpack in the future though, because nobody seems to care that dApps can be as auditable as Picoswap)

**Comment by roschler** [comment]
> because nobody seems to care that dApps can be as auditable as Picoswap)

I do!  If you do move to react/webpack, please keep a pristine branch in its current form.  It's not just auditability, its also having a "universal" code base that is easy to follow that isn't library dependent.  I avoid react/webpack like the plague because my web sites are almost always heavy on the algorithmic side and light on the UI side.  I avoid webpack, at least during 80% of the pre-production phase, because I like knowing the path from my mental image of the code to what code is actually running isn't wrong due to some file or module variance due to webpack.  In fact, when I do most browser dev I check the "disable caching" button.

Either way, a big thank you for this tightly contained project that covers the important bases of a Cardamo client side dApp.

**Comment by roschler** [comment]
Follow-up comment on our conversation about the value of a minimalist code base like Picoswap.  I just had a concrete reminder as to why code bases like this are _critical_.  In another issue I just opened I had an "unexpected error" pop up.  To find the code that was throwing it, all I had to do was use the `pause on exceptions` button with the `pause on caught exceptions` checkbox in Chrome DevTools, and I ended upon the _exact_ line of code and call context where the error was thrown inside the `Helios` library.

I could not have done this is this code base was weighted down with React or any other large library because I would have to wade through tons of caught exceptions, usually with some of them being inside recurring loops or perhaps async code.  Therefore, I would have lost hours trudging through debug traces to find the source of the error.
